### ISSUE TITLE
Support for filtering out subtree(s) when calculating stats

### ISSUE THREAD

**Issue by ricab** [o2sh/onefetch#178]
Just want to suggest an addition to this very nice project. I would find it useful if onefetch had an option to filter out directories and/or files when calculating stats like Languages, LoC and Size. 

I find that current results are quite skewed for projects that include many submodules or other code dependencies. An exclusion pattern would allow ignoring things like third-party code, documentation, examples, etc. I imagine something like `--exclude <pattern>`. 

Alternatively (or additionally), there could be an option to discount git submodules. That would be easier to use, but less general.

**Comment by o2sh** [comment]
Yes, the `--exclude <pattern>` flag would be a nice addition to the tool. [tokei](https://github.com/XAMPPRocky/tokei) (used by onefetch to retrieve language statistics) already has this feature. So we "just" need to use it (from [lib.rs](https://github.com/XAMPPRocky/tokei/blob/master/src/lib.rs)): 

```
// Exclude any path that contains any of these strings.
let excluded = &["target"];
let config = Config::default();
let mut languages = Languages::new();
languages.get_statistics(paths, excluded, &config);
let rust = &languages[&LanguageType::Rust];
```
Doing so, **Languages** and **LoC** sections will be updated based on the flag's input. However the **Size** will remain unchanged as it relies on a git command and not tokei.

**Comment by o2sh** [comment]
https://github.com/XAMPPRocky/tokei/issues/564
